recipes/mysql-connector-c/all/conanfile.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from conans import ConanFile, CMake, tools
+import os
+
+
+
+class MysqlConnectorCConan(ConanFile):
+    name = "mysql-connector-c"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "A MySQL client library for C development."
+    topics = ("conan", "mysql", "sql", "connector", "database")
+    homepage = "https://dev.mysql.com/downloads/connector/c/"
+    author = "Bincrafters <bincrafters@gmail.com>"
+    license = "GPL-2.0"
+    exports = ["LICENSE.md"]
+    exports_sources = ["CMakeLists.txt", "patches/*.patch"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "with_ssl": [True, False], "with_zlib": [True, False]}
+    default_options = {'shared': False, 'with_ssl': True, 'with_zlib': True}
+    _source_subfolder = "source_subfolder"
+
+    def requirements(self):
+        if self.options.with_ssl:
+            self.requires.add("openssl/1.0.2s")
+
+        if self.options.with_zlib:
+            self.requires.add("zlib/1.2.11")
+
+    def source(self):
+        archive_name = self.name + "-" + self.version + "-src"
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(archive_name, self._source_subfolder)
+
+        sources_cmake = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        sources_cmake_orig = os.path.join(self._source_subfolder, "CMakeListsOriginal.txt")
+
+        os.rename(sources_cmake, sources_cmake_orig)
+        os.rename("CMakeLists.txt", sources_cmake)
+        
+        for patch in self.conan_data["patches"][self.version]:
```

Comment:
I prefer applying patches on build stage only. My personal karma is "keep source folder untouched". But I see no problem keeping it here.

Reply:
the only risk there is with your approach is that it breaks when no_copy_source is set

Reply:
yeah, but also when you want to inspect the source code for patches or broken build. Patching can create false-positive